pkg/api/v2/reporter/reporter.go

```diff
@@ -128,6 +131,21 @@ func (e ResourceReports) AddWarning(res resources.InputResource, warning string)
 	e[res] = rpt
 }
 
+func (e ResourceReports) AddMessages(res resources.InputResource, message ...string) {
+	for _, warn := range message {
+		e.AddMessages(res, warn)
+	}
+}
+
+func (e ResourceReports) AddMessage(res resources.InputResource, message string) {
+	if message == "" {
+		return
+	}
+	rpt := e[res]
+	rpt.Messages = append(rpt.Warnings, message)
+	e[res] = rpt
```

Comment:
do we really need the extra copy here? we should be able to append the message to the list in-place